test_runner/fixtures/neon_fixtures.py

```diff
@@ -718,6 +727,26 @@ def enable_real_s3_remote_storage(self, test_name: str, force_enable: bool = Tru
             prefix_in_bucket=self.remote_storage_prefix,
         )
 
+    def cleanup_local_storage(self):
+        if not self.preserve_database_files:
+            directories_to_clean: List[Path] = []
+            for test_entry in Path(self.repo_dir).glob("**/*"):
+                if test_entry.is_file():
+                    test_file = test_entry
+                    if not ATTACHMENT_NAME_REGEX.fullmatch(
+                        test_file.name
+                    ) and not SMALL_DB_FILE_NAME_REGEX.fullmatch(test_file.name):
+                        log.debug(f"Removing large database {test_file} file")
+                        test_file.unlink()
+                elif test_entry.is_dir():
+                    directories_to_clean.append(test_entry)
+
+            while directories_to_clean:
+                directory_to_clean = directories_to_clean.pop()
```

Comment:
```suggestion
            for directory_to_clean in directories_to_clean:
```
I think we don't delete the object itself, so it should be ok (but I haven't tried it)

Reply:
I think I will ignore this one since: we have to remove the directories in the reverse order, to remove the empty deepest ones first, potentially making their parent dirs also empty.

Reply:
Ah, sorry, I missed this part. Maybe then:

```python
for directory_to_clean in directories_to_clean.reverse():
```

Reply:
Sure, thank you for the review!